src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -22,16 +22,16 @@ internal sealed class ForEachLoopBinder : LoopBinder
         private const string CurrentPropertyName = WellKnownMemberNames.CurrentPropertyName;
         private const string MoveNextMethodName = WellKnownMemberNames.MoveNextMethodName;
 
-        private readonly ForEachStatementSyntax _syntax;
+        private readonly CommonForEachStatementSyntax _syntax;
         private SourceLocalSymbol IterationVariable
         {
             get
             {
-                return _syntax.IsDeconstructionDeclaration ? null : (SourceLocalSymbol)this.Locals[0];
+                return (_syntax is ForEachStatementSyntax) ? (SourceLocalSymbol)this.Locals[0] : null;
```

Comment:
Yes, good idea.

---
In reply to: [73615728](https://github.com/dotnet/roslyn/pull/12917#discussion_r73615728) [](ancestors = 73615728)